python/bifrost/block.py

```diff
@@ -275,37 +284,44 @@ def read(self, *args):
             for spans in self.izip(*[sequence.read(self.gulp_size[ring_name]) \
                     for ring_name, sequence in self.izip(args, sequences)]):
                 yield tuple([span.data_view(dtypes[ring_name])[0] for span, ring_name in self.izip(spans, args)])
-                #yield [span.data_view(np.float32)[0] for span in spans]
     def write(self, *args):
         """Iterate over selection of output rings"""
-        # resize all rings
-        for ring_name in args:
-            self.rings[ring_name].resize(self.gulp_size[ring_name])
         # list of sequences
         #TODO: Change this code if someone gives a reasonable answer on 
         #http://stackoverflow.com/questions/38834827/multiple-with-statements-in-python-2-7-using-a-list-comprehension
         with nested(*[self.rings[ring_name].begin_writing() \
                 for ring_name in args]) as out_rings:
-            with nested(*[out_ring.begin_sequence(
-                "",
-                0,
-                header=json.dumps(self.header[ring_name]),
-                nringlet=1) \
-                    for out_ring, ring_name in self.izip(out_rings, args)]) as out_sequences:
-                while True:
-                    with nested(*[out_sequence.reserve(self.gulp_size[ring_name]) \
-                            for out_sequence, ring_name in self.izip(
-                                out_sequences,
-                                args)]) as out_spans:
-                        dtypes = {}
-                        for ring_name in args:
-                            try:
-                                dtype = np.dtype(self.header[ring_name]['dtype']).type
-                            except TypeError:
-                                numpy_dtype_word = self.header[ring_name]['dtype'].split()[1]
-                                dtype = np.dtype(numpy_dtype_word.split(".")[1].split("'")[0]).type
-                            dtypes[ring_name] = dtype
-                        yield tuple([out_span.data_view(dtypes[ring_name])[0] for out_span, ring_name in self.izip(out_spans, args)])
+
+            while True:
+                # resize all rings
+                for ring_name in args:
+                    self.rings[ring_name].resize(self.gulp_size[ring_name])
+
+                with nested(*[out_ring.begin_sequence(
+                    str(int(time.time()*1000)),
+                    int(time.time()*1000),
+                    header=json.dumps(self.header[ring_name]),
+                    nringlet=1) \
+                        for out_ring, ring_name in self.izip(out_rings, args)]) as out_sequences:
+
+                    self.trigger_sequence = False
+                    #TODO: Eventually this could be used on each ring individually.
+                    while not self.trigger_sequence:
```

Comment:
I have added it. Let me know if you think the rest looks good.

Cheers,
Miles